plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/DeltaLakeTableMetadataScheduler.java

```diff
@@ -0,0 +1,258 @@
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        if (enabled) {
+            executor = storeTableMetadataThreads == 0 ? newDirectExecutorService() : newFixedThreadPool(storeTableMetadataThreads, threadsNamed("store-table-metadata-%s"));
+            scheduler = newSingleThreadScheduledExecutor(daemonThreadsNamed("store-table-metadata"));
+
+            scheduler.scheduleWithFixedDelay(() -> {
+                try {
+                    process();
+                }
+                catch (Throwable e) {
+                    log.warn(e, "Error storing table metadata");
+                }
+                try {
+                    checkFailedTasks();
+                }
+                catch (Throwable e) {
+                    log.warn(e, "Error canceling metadata update tasks");
+                }
+            }, 200, 1000, MILLISECONDS);
+        }
+    }
+
+    @VisibleForTesting
+    public void process()
+    {
+        List<Callable<Void>> tasks = new ArrayList<>();
+        synchronized (this) {
+            if (updateInfos.isEmpty()) {
+                return;
+            }
+
+            Map<SchemaTableName, TableUpdateInfo> updateTables = updateInfos.entrySet().stream()
+                    .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue, maxBy(comparing(TableUpdateInfo::version))));
+
+            log.debug("Processing %s table(s): %s", updateTables.size(), updateTables.keySet());
+            for (Map.Entry<SchemaTableName, TableUpdateInfo> entry : updateTables.entrySet()) {
+                tasks.add(() -> {
+                    updateTable(entry.getKey(), entry.getValue());
+                    return null;
+                });
+            }
+
+            this.updateInfos.clear();
+        }
+
+        try {
+            executor.invokeAll(tasks).forEach(MoreFutures::getDone);
+        }
+        catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void updateTable(SchemaTableName schemaTableName, TableUpdateInfo info)
+    {
+        log.debug("Updating table: '%s'", schemaTableName);
+        try {
+            tableOperationsProvider.createTableOperations(info.session)
+                    .commitToExistingTable(schemaTableName, info.version, info.schemaString, info.tableComment);
+            log.debug("Replaced table: '%s'", schemaTableName);
+        }
+        catch (TableNotFoundException e) {
+            // Don't increment failedCounts. The table might have been dropped concurrently.
+            log.debug("Table disappeared during metadata updating operation: '%s'", schemaTableName);
+        }
+        catch (Exception e) {
+            log.warn(e, "Failed to store table metadata for '%s'", schemaTableName);
+            // TODO Consider increment only when the exception is permission issue
+            failedCounts.incrementAndGet();
```

Comment:
This increment strategy feels pretty optimistic - in any case, it can be improved in a follow-up.